projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1122,6 +1115,23 @@ function get_paywall_blocks( $newsletter_access_level ) {
 ';
 }
 
+/**
+ * Returns true if user is auth for subscriptions check, otherwise returns false.
+ *
+ * @return boolean
+ */
+function is_user_auth() {
+	if ( is_user_logged_in() && ( new Host() )->is_wpcom_simple() ) {
+		return true;
+	}
+	if ( class_exists( 'Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Jetpack_Token_Subscription_Service' ) ) {
+		if ( Jetpack_Token_Subscription_Service::has_token_from_cookie() ) {
```

Comment:
Does `has_token_from_cookie` take into account token in the URL, https://github.com/Automattic/jetpack/pull/33761/files?

Reply:
We're moving out of the token in URL. Source of truth is in the header. If it is in the URL, [it should be set early](https://github.com/Automattic/jetpack/blob/1fbf6b7ef3bbb8e3ccf98f7fcc3dfe8f9510f3cb/projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php#L38) in the $_COOKIE.